File: public/docs/_examples/server-communication/dart/lib/wiki/wikipedia_service_1.dart
// #docregion

import 'dart:async';

import 'package:angular2/core.dart';
import 'package:jsonpadding/jsonpadding.dart';

@Injectable()
class WikipediaService {
  Jsonp _jsonp;

  WikipediaService(this._jsonp);

  Future<List<String>> search(String term) async {
    // #docregion query-string
    final queryString = '?search=${term}&action=opensearch&format=json&callback=JSONP_CALLBACK';

    final uri = 'http://en.wikipedia.org/w/api.php' + queryString;

    // TODO: Error handling
    List result = await _jsonp.get(uri);
    return result[1];
    // #enddocregion query-string
  }
}

File: public/docs/_examples/server-communication/ts/app/wiki/wikipedia.service.1.ts
// Create the query string by hand
// #docregion
import { Injectable } from '@angular/core';
import { Jsonp }      from '@angular/http';

@Injectable()
export class WikipediaService {
  constructor(private jsonp: Jsonp) { }

  // TODO: Add error handling
  search(term: string) {

    let wikiUrl = 'http://en.wikipedia.org/w/api.php';

    // #docregion query-string
    let queryString =
      `?search=${term}&action=opensearch&format=json&callback=JSONP_CALLBACK`;

    return this.jsonp
               .get(wikiUrl + queryString)
               .map(request => <string[]> request.json()[1]);
    // #enddocregion query-string
  }
}
